lib/rules/no-restricted-syntax.js

```diff
@@ -10,6 +10,8 @@
 
 module.exports = {
     meta: {
+        type: "style",
```

Comment:
Similar to no-restricted-properties, etc., users could be using this for actual issues. I think this should be a "suggestion" (it's not purely stylistic).